### ISSUE TITLE
Accessibility: change position using keyboard

### ISSUE THREAD

**Issue by dmitry** [zillow/react-slider#41]
It would be nice to have possibility to change position of slider cursors using keyboard.

**Comment by qwtel** [comment]
I'm not sure how I would do that. 

The component would have to render a focusable element, detect the focus and then register keyboard handlers for the duration of the focus. Not sure if that's possible.

Personally I'm using two `<input type="number"/>` elements with the slider that share the same values. Maybe that's a workable solution for you as well.

**Comment by dmitry** [comment]
@cell303 but can you control the values with the arrows (left/right)? Using the slider (with two values) for displaying the price.

**Comment by qwtel** [comment]
Well you would control the values either using the slider or the input fields, which in turn have keyboard listeners (for the up and down arrow keys though).

By using the slider and the input fields as [controlled components](https://facebook.github.io/react/docs/forms.html#controlled-components) you can keep all the values in sync:

``` js
render() {
  const [value1, value2] = this.state.value;
  return (
    <div>
      <Slider value={[value1, value2]} onChange={this.onChange}/>
      <input type="number" value={value1} onChange={v => this.onChange([v, value2])}/>
      <input type="number" value={value2} onChange={v => this.onChange([value1, v])}/>
    </div>
  )
},

onChange(value) {
  this.setState({value});
}
```

**Comment by tacomanator** [comment]
Hopefully a side effect of making it work with the keyboard would be that it is focusable. Right now it's skipped when tabbing, which even users who don't need accessibility features often do to work through a form.

**Comment by imbrianj** [comment]
In addition to merely making it accept focus and keyboard inputs, aria values for current value, min and max should be relatively easily added and offer that functionality to screen-readers.  The `aria-role` of `slider` offers guidance on best practices for a lot of this: https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques/Using_the_slider_role

**Comment by teameh** [comment]
@qwtel your solution works very well for me! I've adapted it a bit:

``` js
<input
  type="number"
  min={min}
  max={max}
  value={value1}
  onChange={e => this.onChange([parseInt(e.currentTarget.value, 10), value2])}
  onKeyDown={(e) => {
    // key right or left
    if (e.keyCode === 37) {
      this.onChange([value1 - 1, value2]);
    } else if (e.keyCode === 39) {
      this.onChange([value1 + 1, value2]);
    }
  }}
/>
<input
  type="number"
  min={min}
  max={max}
  value={value2}
  onChange={e => this.onChange([value1, parseInt(e.currentTarget.value, 10)])}
  onKeyDown={(e) => {
    // key right or left
    if (e.keyCode === 37) {
      this.onChange([value1, value2 - 1]);
    } else if (e.keyCode === 39) {
      this.onChange([value1, value2 + 1]);
    }
  }}
/>
```

To parse the values and also support key left and right events for horizontal sliders :)
But has the downside that I now have to implement the "pearling" and min/max checks as well. 

If would be awesome if this is embedded in the lib along with the right aria attributes @imbrianj mentions.

**Comment by jayphelps** [comment]
WAI-ARIA has a guide for making sliders (including ones with multiple handles, called "thumbs"): https://www.w3.org/TR/wai-aria-practices/#slidertwothumb

There are even reference implementations: http://www.oaa-accessibility.org/examplep/slider1/

tl;dr don't use an input, use `tabindex="0"` w/ all the other necessary aria attributes and then the JS listeners for keydown stuff.

**Comment by teameh** [comment]
Yeah that would be an even better solution! 👍

**Comment by B-Reif** [comment]
I've created a pull request to add some basic keyboard accessibility ([#113](https://github.com/mpowaga/react-slider/pull/113)). I'd appreciate it if we could review this PR.

**Comment by jlfalcon3589** [comment]
WAI-ARIA has a guide to make sliders (https://github.com/mpowaga/react-slider/issues/41#issuecomment-258278570), currently does not support events for the Page Up, Page Down, Home and End key.

Making a wrapper of the component I can handle events, but I do not know which handle I have active when increasing or decreasing the value.

In what way could I know which driver I have active, or if there is a way to handle those events without a wrapper?
